scripts/Connector/Connector_JST/conn_jst_XA_vertical.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = "XA"
+manufacturer = 'JST'
+orientation = 'V'
+number_of_rows = 1
+datasheet = 'http://www.jst-mfg.com/product/pdf/eng/eXA1.pdf'
+
+
+fab_pin1_marker_type = 1
+pin1_marker_offset = 0.3
+pin1_marker_linelen = 1.25
+
+drill_size = 0.95 #Datasheet: 0.9 +0.1/-0
+mh_drill = 1.25 #for boss
+pad_to_pad_clearance = 0.8
+pad_copper_y_solder_length = 0.5 #How much copper should be in y direction?
+min_annular_ring = 0.15
+
+
+
+pitch = 2.5
+
+# as of 2020 : available in 2- to 15-pin, and 20-pin variants.
+pincounts = [*range(2,15+1),20]
```

Comment:
Hmm, that table is facing the "N type" drawings; I took my information from the previous page. To this day I'm not sure if the only difference between "N type" and the other is the note at the bottom of that table, "Wafer : Glass-filled PA66, UL94V-0, natural (ivory)" and the N-type is "natural (white)". Any clue what else is different ?

Reply:
i was asking myself the same thing today, i found [this](https://jsttac.blogspot.com/2011/08/what-is-difference-in-xa-series-header.html).
seems to really be just a different kind of plastic.

Reply:
Haha good find ! I wouldn't have guessed re "unpleasant odor". I have a bag of regulars and can barely smell anything.
Since there's no mechanical difference, I'll just keep both 18- and 20-pin variants. This way the script can cover both standard and N types, and it'll be up to the users to validate their orderable part#  vs pincount. Sounds reasonable ?

Reply:
sounds good. just make sure the script doesn't generate an 18 pin connector _with_ boss, too